OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs

```diff
@@ -234,8 +269,12 @@ public IEnumerable<IRenderable> RenderAboveShroud(WorldRenderer wr, World world)
 
 				var res = world.WorldActor.TraitOrDefault<ResourceLayer>();
 				var isCloseEnough = buildingInfo.IsCloseEnoughToBase(world, world.LocalPlayer, building, topLeft);
+				var replacmentInfo = world.Map.Rules.Actors[building].TraitInfoOrDefault<ReplacementInfo>();
 				foreach (var t in buildingInfo.Tiles(topLeft))
-					cells.Add(t, MakeCellType(isCloseEnough && world.IsCellBuildable(t, buildingInfo) && (res == null || res.GetResource(t) == null)));
+					cells.Add(t, MakeCellType(isCloseEnough
+						&& (world.IsCellBuildable(t, buildingInfo)
+						|| ((replacmentInfo == null) ? false : AcceptsReplacement(t, replacmentInfo)))
+						&& (res == null || res.GetResource(t) == null)));
 			}
```

Comment:
I am aware of a style violation 
`OpenRA.Mods.Common\Orders\PlaceBuildingOrderGenerator.cs:L274: [SA1118] The parameter spans multiple lines. If the parameter is short, place the entire parameter on a single line. Otherwise, save the contents of the parameter in a temporary variable and pass the temporary variable as a parameter.`
Should i shorten it to one line? Dont think a temporary variable is helpful here